shell/common/platform_view_service_protocol.cc

```diff
@@ -314,4 +316,29 @@ void PlatformViewServiceProtocol::ScreenshotGpuTask(SkBitmap* bitmap) {
   canvas->flush();
 }
 
+const char* PlatformViewServiceProtocol::kWaitUIThreadIdleExtensionName =
+    "_flutter.waitUIThreadIdle";
```

Comment:
Strictly speaking this is not waiting for the UI thread to idle it is more wait for UI thread to finish processing the first event, I don't have a better name for it though.